--- ./libopie/Makefile.in.orig	Wed Aug  8 23:39:35 2001
+++ ./libopie/Makefile.in	Wed Aug  8 23:44:17 2001
@@ -19,13 +19,25 @@
 CC=@CC@
 CFLAGS=$(CFL) -I.. -I../libmissing
 TARGET=libopie.a
+TARGETS=libopie.so
+MAJOR=2
+MINOR=4
 
-all: $(TARGET)
+all: $(TARGET) $(TARGETS)
 
 $(TARGET): $(OBJS)
 	ar cr $(TARGET) $(OBJS)
 	@RANLIB@ $(TARGET)
 
+$(TARGETS): $(OBJS)
+	ld -o $(TARGETS).$(MAJOR).$(MINOR) -soname $(TARGETS).$(MAJOR) -shared $(OBJS)
+	ln -s $(TARGETS).$(MAJOR).$(MINOR) $(TARGETS)
+
+install: $(TARGET) $(TARGETS)
+	install $(TARGET) $(DESTDIR)$(LOCALLIB)
+	install $(TARGETS).$(MAJOR).$(MINOR) $(DESTDIR)$(LOCALLIB)
+	ln -s $(TARGETS).$(MAJOR).$(MINOR) $(DESTDIR)$(LOCALLIB)/$(TARGETS)
+
 clean:
 	-rm -f $(OBJS) $(TARGET)
 
--- ./configure.in.orig	Thu Mar 11 03:09:53 1999
+++ ./configure.in	Wed Aug  8 23:07:59 2001
@@ -460,7 +460,8 @@
 
 dnl Checks for libraries.
 AC_CHECK_LIB(crypt, crypt)
-AC_CHECK_LIB(nsl, gethostname)
+AC_CHECK_FUNC(gethostbyname, ,
+	AC_CHECK_LIB(nsl, gethostname)) 
 AC_CHECK_LIB(posix, main)
 AC_CHECK_LIB(socket, socket)
 
--- ./Makefile.in.orig	Wed Aug  8 23:13:14 2001
+++ ./Makefile.in	Thu Aug  9 00:40:16 2001
@@ -113,6 +113,8 @@
 MKDIR=@MKDIR@
 CHOWN=@CHOWN@
 LOCALBIN=@LOCALBIN@
+LOCALSBIN=@LOCALSBIN@
+LOCALLIB=@libdir@
 LOCALMAN=@LOCALMAN@
 SU=@SU@
 ALT_SU=@ALT_SU@
@@ -132,123 +134,41 @@
 ifdefs:	
 	egrep '^#*if*def' *.c *.h | cut -f2 -d: | sort | uniq
 
-client: libopie/libopie.a libmissing/libmissing.a opietest-passed opiekey opiegen $(OPIEAUTO)
+client: libopie/libopie.so libopie/libopie.a libmissing/libmissing.a opietest-passed opiekey opiegen $(OPIEAUTO)
 
 client-install: client
 	@echo "Installing OPIE client software..."
 	@echo "Copying OPIE key-related files"
-	@if test ! -d $(LOCALBIN); then $(MKDIR) $(LOCALBIN); chmod 755 $(LOCALBIN); fi 
-	@cp opiekey $(OPIEAUTO) $(LOCALBIN)
-	@$(CHOWN) $(OWNER) $(LOCALBIN)/opiekey
-	@if test ! -z "$(OPIEAUTO)"; then $(CHOWN) $(OWNER) $(LOCALBIN)/opieauto; fi
-	@chgrp $(GROUP) $(LOCALBIN)/opiekey
-	@echo "Changing file permissions"
-	@chmod 0511 $(LOCALBIN)/opiekey
-	@if test ! -z "$(OPIEAUTO)"; then chmod 0511 $(LOCALBIN)/opieauto; fi
+	@if test ! -d $(DESTDIR)$(LOCALBIN); then $(MKDIR) $(DESTDIR)$(LOCALBIN); fi
+	@cp opiekey $(DESTDIR)$(LOCALBIN)
 	@echo "Symlinking aliases to opiekey"
-	@-ln -s $(LOCALBIN)/opiekey $(LOCALBIN)/otp-md4
-	@-ln -s $(LOCALBIN)/opiekey $(LOCALBIN)/otp-md5
+	@-ln -s opiekey $(DESTDIR)$(LOCALBIN)/otp-md4
+	@-ln -s opiekey $(DESTDIR)$(LOCALBIN)/otp-md5
 	@echo "Installing manual pages"
-	@-for i in otp-md4 otp-md5; do ln -s opiekey.1 $(LOCALMAN)/man1/$$i.1; done
-	@if test ! -d $(LOCALMAN)/man1; then $(MKDIR) $(LOCALMAN)/man1; chmod 755 $(LOCALMAN)/man1; fi; cp opiekey.1 $(LOCALMAN)/man1/opiekey.1; $(CHOWN) $(OWNER) $(LOCALMAN)/man1/opiekey.1; chgrp $(GROUP) $(LOCALMAN)/man1/opiekey.1; chmod 644 $(LOCALMAN)/man1/opiekey.1
+	@-for i in otp-md4 otp-md5; do ln -s opiekey.1 $(DESTDIR)$(LOCALMAN)/man1/$$i.1; done
+	@if test ! -d $(DESTDIR)$(LOCALMAN)/man1; then $(MKDIR) $(DESTDIR)$(LOCALMAN)/man1; fi;
+	@-for i in otp-md4 otp-md5; do echo ".so opiekey.1" > $(DESTDIR)$(LOCALMAN)/man1/$$i.1; done
 
-server:	libopie/libopie.a libmissing/libmissing.a opietest-passed opielogin opiesu opiepasswd opieinfo opieftpd opieserv
+server:	libopie/libopie.a libopie/libopie.so libmissing/libmissing.a opietest-passed opielogin opiesu opiepasswd opieinfo opieftpd opieserv
 
 server-install: server
 	@echo "Installing OPIE server software..."
 	@echo "Copying OPIE user programs"
-	@if test ! -d $(LOCALBIN); then $(MKDIR) $(LOCALBIN); chmod 755 $(LOCALBIN); fi 
-	@cp opiepasswd opieinfo $(LOCALBIN)
-	@echo "Changing ownership"
-	@$(CHOWN) $(OWNER) $(LOCALBIN)/opiepasswd $(LOCALBIN)/opieinfo
-	@chgrp $(GROUP) $(LOCALBIN)/opiepasswd $(LOCALBIN)/opieinfo
-	@echo "Changing file permissions"
-	@chmod 0555 $(LOCALBIN)/opieinfo
-	@chmod 4511 $(LOCALBIN)/opiepasswd
+	@if test ! -d $(DESTDIR)$(LOCALBIN); then $(MKDIR) $(DESTDIR)$(LOCALBIN); fi 
+	@cp opiepasswd opieinfo $(DESTDIR)$(LOCALBIN)
 	@echo "Installing OPIE system programs..."
-	@if test ! -z $(LOGIN); \
-	then \
-		if test ! $(EXISTS) $(LOGIN).$(BACKUP); \
-		then \
-			echo "Renaming existing $(LOGIN) to $(LOGIN).$(BACKUP)"; \
-			mv $(LOGIN) $(LOGIN).$(BACKUP); \
-			echo "Clearing permissions on $(LOGIN)"; \
-			chmod 0 $(LOGIN).$(BACKUP); \
-		fi; \
-		echo "Copying OPIE login to $(LOGIN)"; \
-		cp opielogin $(LOGIN); \
-		echo "Changing ownership of $(LOGIN)"; \
-		$(CHOWN) $(OWNER) $(LOGIN); \
-		chgrp $(GROUP) $(LOGIN); \
-		echo "Changing file permissions of $(LOGIN)"; \
-		chmod 4111 $(LOGIN); \
-	fi
-	@if test ! -z $(SU); \
-	then \
-		if test ! $(EXISTS) $(SU).$(BACKUP); \
-		then \
-			echo "Renaming existing $(SU) to $(SU).$(BACKUP)"; \
-			mv $(SU) $(SU).$(BACKUP); \
-			echo "Clearing permissions on $(SU)"; \
-			chmod 0 $(SU).$(BACKUP); \
-		fi; \
-		echo "Copying OPIE su to $(SU)"; \
-		cp opiesu $(SU); \
-		echo "Changing ownership of $(SU)"; \
-		$(CHOWN) $(OWNER) $(SU); \
-		chgrp $(GROUP) $(SU); \
-		echo "Changing file permissions of $(SU)"; \
-		chmod 4111 $(SU); \
-	fi
-	@if test ! -z $(ALT_SU); \
-	then \
-		if test ! $(EXISTS) $(ALT_SU).$(BACKUP); \
-		then \
-			echo "Renaming existing $(ALT_SU) to $(ALT_SU).$(BACKUP)"; \
-			mv $(ALT_SU) $(ALT_SU).$(BACKUP); \
-			echo "Clearing permissions on $(ALT_SU)"; \
-			chmod 0 $(ALT_SU).$(BACKUP); \
-		fi; \
-		echo "Copying OPIE su to $(ALT_SU)"; \
-		cp opiesu $(ALT_SU); \
-		echo "Changing ownership of $(ALT_SU)"; \
-		$(CHOWN) $(OWNER) $(ALT_SU); \
-		chgrp $(GROUP) $(ALT_SU); \
-		echo "Changing file permissions of $(ALT_SU)"; \
-		chmod 4111 $(ALT_SU); \
-	fi
-	@if test ! -z $(FTPD); \
-	then \
-		if test ! $(EXISTS) $(FTPD).$(BACKUP); \
-		then \
-			echo "Renaming existing $(FTPD) to $(FTPD).$(BACKUP)"; \
-			mv $(FTPD) $(FTPD).$(BACKUP); \
-			echo "Clearing permissions on $(FTPD).$(BACKUP)"; \
-			chmod 0 $(FTPD).$(BACKUP); \
-		fi; \
-		echo "Copying OPIE ftp daemon to $(FTPD)"; \
-		cp opieftpd $(FTPD); \
-		echo "Changing ownership of $(FTPD)"; \
-		$(CHOWN) $(OWNER) $(FTPD); \
-		chgrp $(GROUP) $(FTPD); \
-		echo "Changing file permissions of $(FTPD)"; \
-		chmod 0100 $(FTPD); \
-	fi
+	@cp opieftpd opiesu opielogin $(DESTDIR)$(LOCALSBIN)
 	@echo "Making sure OPIE database file exists";
-	@touch $(KEY_FILE)
-	@echo "Changing permissions of OPIE database file"
-	@chmod 0644 $(KEY_FILE)
-	@echo "Changing ownership of OPIE database file"
-	@$(CHOWN) $(OWNER) $(KEY_FILE)
-	@chgrp $(GROUP) $(KEY_FILE)
-	@-if test ! -z "$(LOCK_DIR)"; then echo "Creating OPIE lock directory"; mkdir $(LOCK_DIR); $(CHOWN) 0 $(LOCK_DIR); chgrp 0 $(LOCK_DIR); chmod 0700 $(LOCK_DIR); fi;
-	@-if test ! -z "$(ACCESS_FILE)"; then echo "Creating OPIE access file (don't say we didn't warn you)"; touch $(ACCESS_FILE); $(CHOWN) 0 $(ACCESS_FILE); chgrp 0 $(ACCESS_FILE); chmod 0444 $(ACCESS_FILE); fi;
+	@touch $(DESTDIR)$(KEY_FILE)
+	@-if test ! -z "$(DESTDIR)$(LOCK_DIR)"; then echo "Creating OPIE lock directory"; mkdir $(DESTDIR)$(LOCK_DIR); fi;
+	@-if test ! -z "$(DESTDIR)$(ACCESS_FILE)"; then echo "Creating OPIE access file (don't say we didn't warn you)"; touch $(DESTDIR)$(ACCESS_FILE); fi;
 	@echo "Installing manual pages"
-	@if test ! -d $(LOCALMAN); then $(MKDIR) $(LOCALMAN); chmod 755 $(LOCALMAN); fi
-	@for i in 1 4 5 8; do for j in *.$$i; do if test ! -d $(LOCALMAN)/man$$i; then $(MKDIR) $(LOCALMAN)/man$$i; chmod 755 $(LOCALMAN)/man$$i; fi; cp $$j $(LOCALMAN)/man$$i/$$j; $(CHOWN) $(OWNER) $(LOCALMAN)/man$$i/$$j; chgrp $(GROUP) $(LOCALMAN)/man$$i/$$j; chmod 644 $(LOCALMAN)/man$$i/$$j; done; done
+	@if test ! -d $(DESTDIR)$(LOCALMAN); then $(MKDIR) $(DESTDIR)$(LOCALMAN); fi
+	@for i in 1 4 5 8; do for j in *.$$i; do if test ! -d $(DESTDIR)$(LOCALMAN)/man$$i; then $(MKDIR) $(DESTDIR)$(LOCALMAN)/man$$i; fi; cp $$j $(DESTDIR)$(LOCALMAN)/man$$i/$$j; done; done
 	@echo "REMEMBER to run opiepasswd on your users immediately."
 
 install: client-install server-install
+	(cd libopie ; $(MAKE) install LOCALLIB=$(LOCALLIB))
 
 uninstall:
 	@echo "Un-installing OPIE..."
@@ -269,11 +189,14 @@
 	@echo "Please verify by hand that this process worked."
 
 opietest-passed:	opietest
-	-./opietest && touch opietest-passed
+	LD_PRELOAD=libopie/libopie.so ./opietest && touch opietest-passed
 
 libopie/libopie.a: libopie/*.c *.h
 	(cd libopie ; $(MAKE) libopie.a CFL='$(CFLAGS) -DKEY_FILE=\"$(KEY_FILE)\"')
 
+libopie/libopie.so: libopie/*.c *.h
+	(cd libopie ; $(MAKE) libopie.so CFL='$(CFLAGS) -DKEY_FILE=\"$(KEY_FILE)\"')
+
 libmissing/libmissing.a: libmissing/*.c
 	(cd libmissing ; $(MAKE) libmissing.a CFL='$(CFLAGS)')
 
--- ./configure.munger.in.orig	Wed Aug  8 23:34:35 2001
+++ ./configure.munger.in	Wed Aug  8 23:36:07 2001
@@ -1,9 +1,13 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 bindir=@bindir@
+sbindir=@sbindir@
+libdir=@libdir@
 mandir=@mandir@
 
 LOCALBIN=$bindir
+LOCALSBIN=$sbindir
+LOCALLIB=$libdir
 LOCALMAN=$mandir
 
 echo ""
@@ -11,6 +15,6 @@
 echo "Manual pages are going to be installed into $LOCALMAN."
 echo ""
 echo "creating Makefile"
-cat Makefile.munge | sed s:@LOCALMAN@:$LOCALMAN:g | sed s:@LOCALBIN@:$LOCALBIN:g > Makefile
+cat Makefile.munge | sed s:@LOCALMAN@:$LOCALMAN:g | sed s:@LOCALBIN@:$LOCALBIN:g |sed s:@LOCALSBIN@:$LOCALSBIN:g | sed s:@LOCALBIN@:$LOCALBIN:g > Makefile
 echo ""
 echo "Have you read the README file?"
