diff -Nru opie-2.32/configure.in opie-2.32.new/configure.in
--- opie-2.32/configure.in	Fri Jan  2 00:53:24 1998
+++ opie-2.32.new/configure.in	Tue Aug 22 12:38:14 2000
@@ -453,7 +453,8 @@
 
 dnl Checks for libraries.
 AC_CHECK_LIB(crypt, crypt)
-AC_CHECK_LIB(nsl, gethostname)
+AC_CHECK_FUNC(gethostbyname, ,
+	AC_CHECK_LIB(nsl, gethostname)) 
 AC_CHECK_LIB(posix, main)
 AC_CHECK_LIB(socket, socket)
 
